### ISSUE TITLE
TrainCarts are become normal Minecraft minecarts after server restart

### ISSUE THREAD

**Issue by RichVillage** [bergerhealer/TrainCarts#230]
##### Spigot version: Spigot 1.12.2-R0.1-SNAPSHOT

<hr>

#### Steps to reproduce:
1. Activate traincarts and station. Traincarts start move.
2. Server restart / or server have some lag, traincarts are stop.
3. Check traincarts whether it is traincarts or normal minecart -> They are become normal minecart.

Sorry for my English.. My server have serious problem for this problem.

**Comment by bergerkiller** [comment]
> Server restart / or server have some lag, traincarts are stop.
When the server is force-killed when it gets frozen, traincarts gets no chance to save the latest position information of the managed minecarts to file (groupdata.dat). This would result in minecarts that existed in the world to be disassociated and turn into normal minecarts.

A remedy can be to save groupdata regularly, but that is kind of useless, because minecarts move around. Itll end up losing track of minecarts either way.

Best way to avoid it is to make sure a server doesn't freeze, and do occasional restarts if needed.

**Comment by RichVillage** [comment]
@bergerkiller Yes, I have a server log. Thank you for your reply. But, TrainCart make very very serious lag. TrainCart is not despawn automatically (we have destroy sign).. so server is freezing. You have some solution..?

**Comment by RichVillage** [comment]
@bergerkiller Yes, I am set keep loaded true for all trains by sign property.
I also set false for "keepChunksLoadedOnlyMoving".

<Sign property>
[+train]
keeploaded
true

**Comment by bergerkiller** [comment]
Are you using normal rails or tc-coasters rails? if trains arent being despawned, maybe it helps to make the destroy sign more sensitive by making it trigger from all directions.
```
[+train:nesw]
destroy
```
Other than that I don't really know, sorry. It would help to know why so many trains are spawning that aren't despawned. I would recommend giving your trains names from where they are spawned, so you can later track what kind of train there are many of using /train list
```
[+train]
property
rename
somename#
```

**Comment by RichVillage** [comment]
@bergerkiller My server is not have TC-Coaster Rail, so I think we use normal rails. I will try more. Thank you for your help. Have a nice day!
If I have some changes, please let me comment to you..
